src/v/kafka/server/protocol_utils.h

```diff
@@ -26,4 +26,10 @@ ss::future<std::optional<request_header>> parse_header(ss::input_stream<char>&);
 
 ss::scattered_message<char> response_as_scattered(response_ptr response);
 
+class malformed_header_exception : public std::runtime_error {
```

Comment:
Have you considered making `malformed_header_exception` inherit from `net::parsing_exception`? That seems appropriate and seems to be the status quo approach.

Reply:
To me, it seemed that it is the current intended customisation point for exactly these kinds of "unclean connection abort leads to malformed state" exceptions. It would make the exception logging branching simpler and avoid having two ways of doing seemingly the same thing. But fair point on moving off of `std::runtime_error`, I'm okay with either approach.

Reply:
done